File: app/src/main/java/com/bq/zowi/controllers/BTAdapterController.java
package com.bq.zowi.controllers;

import android.bluetooth.BluetoothDevice;
import rx.Observable;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface BTAdapterController {

    public static class BluetoothNotDisabledException extends IllegalStateException {
    }

    public static class BluetoothNotEnabledException extends IllegalStateException {
    }

    Single<Void> disableBluetooth();

    Single<Void> enableBluetooth();

    BluetoothDevice getRemoteDevice(String str);

    Observable<BluetoothDevice> startBTDiscovery();

    void stopBTDiscovery();
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/BTAdapterController.kt
package com.bq.zowi.api

import io.reactivex.Observable
import io.reactivex.Single

interface BTAdapterController {
    class BluetoothNotDisabledException : IllegalStateException()
    class BluetoothNotEnabledException : IllegalStateException()

    fun disableBluetooth(): Single<Void>
    fun enableBluetooth(): Single<Void>
    fun getRemoteDevice(address: String): DeviceHandle
    fun startBTDiscovery(): Observable<DeviceHandle>
    fun stopBTDiscovery()
}
